builtin/builtin.go

```diff
@@ -826,6 +826,50 @@ object.
 The globals and locals are dictionaries, defaulting to the current
 globals and locals.  If only globals is given, locals defaults to it.`
 
+const isinstance_doc = `isinstance(obj, class_or_tuple) -> bool
+
+Return whether an object is an instance of a class or of a subclass thereof.
+
+A tuple, as in isinstance(x, (A, B, ...)), may be given as the target to
+check against. This is equivalent to isinstance(x, A) or isinstance(x, B)
+or ... etc.
+`
+
+func isinstance(args py.Tuple) (py.Bool, error) {
+	switch args[1].(type) {
+	case py.Tuple:
+		{
+			var res py.Bool
+			var class = args[1].(py.Tuple)
+			for idx := range class {
+				var new_args []py.Object
+				new_args = append(new_args, args[0], class[idx])
+				temp, err := isinstance(new_args)
+				if err != nil {
+					return false, err
+				}
+				res = res || temp
+			}
+			return res, nil
+		}
+	default:
+		{
+			if args[0].Type() != py.TypeType {
+				return false, py.ExceptionNewf(py.TypeError, "isinstance() arg 2 must be a type or tuple of types")
+			}
+			return args[0].Type() == args[1], nil
+		}
+	}
+}
+
+func builtin_isinstance(self py.Object, args py.Tuple) (py.Object, error) {
+	if len(args) > 2 {
+		return nil, py.ExceptionNewf(py.TypeError, "isinstance expected 2 arguments, got %d", len(args))
+	}
+
+	return isinstance(args)
```

Comment:
When you use `err := UnpackTuple(args, nil, "isinstance", 2, 2, obj, classOrTuple)` it will check all cases and you don't have to write your own error message, which would increase binary size.

Reply:
@tim-st I agree with your point. Used UnpackTuple function instead.

Reply:
@xarus01 I think the line `return nil, err` didn't format by your IDE, otherwise this change looks good!